apps/workspace-host/src/interface.ts

```diff
@@ -822,28 +861,32 @@ async function _createContainer(workspace) {
   return container;
 }
 
-async function _startContainer(workspace) {
+async function _startContainer(workspace: Workspace) {
   await workspaceUtils.updateWorkspaceMessage(workspace.id, 'Starting container');
+  if (!workspace.container) {
+    throw new Error('Container not found');
+  }
   await workspace.container.start();
 }
 
 /**
  * Wrapper around `updateWorkspaceState()` that will ensure any errors don't
  * propagate to the caller. Useful during the initialization sequence when we
  * would mark the host as unhealthy if an error occurred while updating the state.
- * @param {*} workspaceId
- * @param {*} state
- * @param {*} message
  */
-function safeUpdateWorkspaceState(workspaceId, state, message) {
+function safeUpdateWorkspaceState(
+  workspaceId: string | number,
+  state: 'stopped',
```

Comment:
👀 is this really the only state that callers ever use?

Reply:
Renamed to `safeStopWorkspace`.

Reply:
I don't think that's a good name. To me, that implies it's going to take some action in the workspace (stop a container or the like), but it just updates the state in the DB (and broadcasts to clients IIRC).

Reply:
Yeah in hindsight that is an awful name. I changed it to `markWorkspaceStopped`.